### PR TITLE
TTL tests and fixes

### PR DESCRIPTION
**Pull request by glaslos**
I did some tests regarding the comment https://github.com/nvie/rq/pull/453#issuecomment-66739929
Seems like the job finishes even thou the job has already expired, no hiccup but we get a result. Is this expected behavior?

### CODE REVIEW

**Comment by selwin** [comment]
That's because the running Python process still has a copy of the job instance which is then saved to Redis again after execution is finished. However, in the cases of hard shutdown or power failure, job data will still be lost. Also, jobs should never expire in the case of failed execution so it can be inspected later. 

So I think we should still set jobs to never expire `job.ttl` right before execution.

**Comment by glaslos** [comment]
Sounds fair to me. I'll try to fix that this week.

**Comment by glaslos** [comment]
@selwin I added the never expire right before performing the job. I fetch the TTL from redis right after that. This could be also set always to -1. redis.ttl() is O(1).

File: rq/job.py [hunk 1]
```diff
@@ -485,6 +485,8 @@ def delete(self, pipeline=None):
     # Job execution
     def perform(self):  # noqa
         """Invokes the job function with the job arguments."""
+        self.connection.persist(self.key)
+        self.ttl = self.connection.ttl(self.key)
```
Comment (selwin) [inline rq/job.py]: What's the intention of this line? Doesn't `connection.ttl` always return `-1` since we called `connection.persist` on the line above?

Comment (glaslos) [inline rq/job.py]: Yeah, see https://github.com/nvie/rq/pull/477#issuecomment-76408703 we could hard-code it to `-1`

Comment (selwin) [inline rq/job.py]: Ah, I missed your comment above. Yes, please hard code it to -1. This saves a round trip to Redis :)

**Comment by selwin** [comment]
Thanks for the PR :)

**Pull request merged**